tasks.c

```diff
@@ -2616,7 +2631,9 @@ BaseType_t xTaskCatchUpTicks( TickType_t xTicksToCatchUp )
     /* Use xPendedTicks to mimic xTicksToCatchUp number of ticks occurring when
      * the scheduler is suspended so the ticks are executed in xTaskResumeAll(). */
     vTaskSuspendAll();
+    taskENTER_CRITICAL();
     xPendedTicks += xTicksToCatchUp;
+    taskEXIT_CRITICAL();

```

Comment:
Same as above

Reply:
I think both critical sections are required in case a tick interrupt occurs while this code is trying to modify `xPendedTicks`.  When the scheduler is suspended, the tick interrupt increments `xPendedTicks`.